### ISSUE TITLE
Training custom object detector using different weights file

### ISSUE THREAD

**Issue by phongnhhn92** [AlexeyAB/darknet#165]
Hello, I have some question:
1. In case of YOLOv2, If i am using the pre-trained weights darknet19_448.conv.23 for training custom object detector, then it means that I just used all the weights from layer 0 to 23 from the pre-trained weights file and from layer 24 to 30, all the weights are initialized from the scratch, is that correct ? I also confused how did those weights from layer 24 to 30 initialized ?
2. If I understand this correctly, then I can train YOLOv2 with different type of backbone architecture. For example, in the ImageNet Classification section [https://pjreddie.com/darknet/imagenet/](url), the author of Darknet provided pretrained models like Resnet. So my question is how to implement them for training custom object detector ?

**Comment by AlexeyAB** [comment]
@phongnhhn92 Hi,

1. Yes. Weights from layer 24 to 30 initialized by random values. This is just initial point, to train faster.
2. For example, 
* download 2 files `resnet50.cfg` and `resnet50.weights`: https://pjreddie.com/darknet/imagenet/
* do `darknet.exe partial resnet50.cfg resnet50.weights resnet50.weights.28 28`
* change first 28 layers in `yolo-voc.2.0.cfg` to 28 layers from`resnet50.cfg`, and get `yolo-voc-resnet28.cfg`
* and train `darknet.exe detector train data/obj.data yolo-voc-resnet28.cfg resnet50.weights.28` as usuall: https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

Or you can change first 28 layers in yolo-voc.2.0.cfg to 48 firts layers from resnet50.cfg.

**Comment by phongnhhn92** [comment]
Thanks for your comment ! I downloaded the resnet cfg and weights files. However, when i tried a simple prediction command then I got this error. 
![capture](https://user-images.githubusercontent.com/11288381/29596257-4b7f7368-87f7-11e7-9051-c19ea3bebc4c.PNG)
However, I have no problem with Dense Net prediction, so I dont know what's wrong in here ? May be i will try to use Dense net as feature extractor.

**Comment by AlexeyAB** [comment]
@phongnhhn92 Yes, try to use Densnet201 instead of Resnet50 - this network conatins only well works layer in Darknet: convolutional, maxpool, avgpool, route, softmax, (cost): https://github.com/pjreddie/darknet/blob/master/cfg/densenet201.cfg

**Comment by phongnhhn92** [comment]
@AlexeyAB I mean I download this weights and cfg file from the official page of Darknet so I guess everything would have to work, right ?

**Comment by AlexeyAB** [comment]
@phongnhhn92 There were bug, I fixed it. Now AlexNet and ResNet50 works fine. Update your code.

Also to use Densenet201 as detector, follow this manual: https://github.com/AlexeyAB/darknet/issues/179#issuecomment-329829708
